modules/images/webp-uploads/load.php

```diff
@@ -704,3 +704,184 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
 	return rest_ensure_response( $data );
 }
 add_filter( 'rest_prepare_attachment', 'webp_uploads_update_rest_attachment', 10, 3 );
+
+/**
+ * Inspect if the current call to `wp_update_attachment_metadata()` was done from within the context
+ * of an edit to an attachment either restore or other type of edit, in that case we perform operations
+ * to save the sources properties, specifically for the `full` size image due this is a virtual image size.
+ *
+ * @since n.e.x.t
+ *
+ * @see wp_update_attachment_metadata()
+ *
+ * @param array $data          The current metadata of the attachment.
+ * @param int   $attachment_id The ID of the current attachment.
+ * @return array The updated metadata for the attachment to be stored in the meta table.
+ */
+function webp_uploads_update_attachment_metadata( $data, $attachment_id ) {
+
+	$trace = debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS, 10 );
```

Comment:
Since this function is intended only for ajax requests with the `image-editor` action, we can return early if we are not doing the `wp_ajax_image-editor` action:
```suggestion
	if ( ! doing_action( 'wp_ajax_image-editor' ) ) {
		return;
	}

	$trace = debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS, 10 );
```

Reply:
Right we might need to update the test to work in that scenario but I agree that this can be safely introduced, let me verify on my end as well.

Reply:
@eugene-manuilov @mitogh I'd prefer we avoid adding that condition. Yes, it's only called by the AJAX function in WordPress right now, but our code should remain agnostic of that if at all possible. If someone calls e.g. `wp_save_image()` in whatever other context, our code here should still do its job.

Reply:
Good point, this would help with tests too as I was spending some time already to pass the tests under this scenario.

Reply:
> If someone calls e.g. `wp_save_image()` in whatever other context, our code here should still do its job.

@felixarntz, I think that it's very unlikely that someone will use this function for their own purposes because that function is tough coupled with the logic implemented in the image editor and heavily relies on various request variables using a lot of `$_REQUEST[...]` calls.

The matter here is more ethical, I think... In some places, we fight for minor improvements to get a super tiny performance gain, in other places we are generous enough to leave a redundant function call that parses the stack trace and generates useless information.

Reply:
@eugene-manuilov I agree in a few PRs so far some discussions were overly nit-picking, let's try to avoid "do we use `array_map` or `foreach`" type of discussions.

But in terms of implementation scope I'd always advocate for maximum compatibility when working with core. Because of the (lack of) core architecture in many places anything can happen, and it's important we don't only consider how core uses a function but also that a plugin could use it in a different context. For example just because a function is only called by core during an AJAX or REST request doesn't mean we should only support that case - the only exception to that would be of course functions that are explicitly named/marked as such (like the `wp_ajax_*()` functions).